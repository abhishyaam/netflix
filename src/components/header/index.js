import React from 'react';
import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSeacrhActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSeacrhActive((searchActive) => !searchActive)}
      >
        <img src='/images/icons/search.png' alt='Search' />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder='Search films and series'
        active={searchActive}
      />
    </Search>
  );
};
