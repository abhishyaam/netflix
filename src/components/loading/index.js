import React from 'react';
import { Picture, Spinner, LockBody, ReleaseBody } from './styles/loading';

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid='loading-picture' />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingRealeaseBody({ src, ...restProps }) {
  return <ReleaseBody />;
};
