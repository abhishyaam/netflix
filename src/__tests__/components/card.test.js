import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Card, Player } from '../../components';

const category = 'series';
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP',
      },
    ],
  },
];

describe('<Card /> ', () => {
  it('renders <Card /> with popluated data', () => {
    const { queryByText, container } = render(
      <Card.Group>
        {slideRows.map((sildeItem) => (
          <Card key={`category-${sildeItem.title.toLowerCase()}`}>
            <Card.Title>{sildeItem.title}</Card.Title>
            <Card.Entities>
              {sildeItem.data.map((item) => (
                <Card.Item key={item.id} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src='/videos/bunny.mp4' />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('Documentaries')).toBeTruthy();
    expect(queryByText('Tiger King')).toBeTruthy();
    expect(queryByText('Tiger King description')).toBeTruthy();
    expect(queryByText('Feel Good')).toBeTruthy();
    expect(queryByText('Juno')).toBeTruthy();
    expect(queryByText('Juno description')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a <Card /> clicks card feature', () => {
    const { queryByText, container, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((sildeItem) => (
          <Card key={`category-${sildeItem.title.toLowerCase()}`}>
            <Card.Title>{sildeItem.title}</Card.Title>
            <Card.Entities>
              {sildeItem.data.map((item) => (
                <Card.Item
                  key={item.id}
                  item={item}
                  data-testid={`${item.slug}-feature`}
                >
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src='/videos/bunny.mp4' />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('18'));
    fireEvent.click(getByTestId('tiger-king-feature'));
    expect(queryByText('18')).toBeTruthy();

    fireEvent.click(getByAltText('close'));
    expect(queryByText('18')).toBeFalsy();

    expect(queryByText('PG')).toBeFalsy();
    fireEvent.click(getByTestId('juno-feature'));
    expect(queryByText('PG')).toBeTruthy();

    fireEvent.click(getByAltText('close'));
    expect(queryByText('PG')).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
