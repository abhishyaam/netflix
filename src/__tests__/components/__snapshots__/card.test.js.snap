// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card />  renders <Card /> with popluated data 1`] = `
<div
  class="card__Group-sc-s87lnr-2 eSwNi"
>
  <div
    class="card__Container-sc-s87lnr-1 krKFpC"
  >
    <p
      class="card__Title-sc-s87lnr-0 gdSujY"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-s87lnr-13 gIQTQQ"
    >
      <div
        class="card__Item-sc-s87lnr-12 iMLVzk"
      >
        <img
          class="card__Image-sc-s87lnr-14 cpMsCx"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 inrCyd"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 igrqGk"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-s87lnr-4 gQlijB"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-s87lnr-1 krKFpC"
  >
    <p
      class="card__Title-sc-s87lnr-0 gdSujY"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-s87lnr-13 gIQTQQ"
    >
      <div
        class="card__Item-sc-s87lnr-12 iMLVzk"
      >
        <img
          class="card__Image-sc-s87lnr-14 cpMsCx"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 inrCyd"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 igrqGk"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-s87lnr-4 gQlijB"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card />  renders a <Card /> clicks card feature 1`] = `
<div
  class="card__Group-sc-s87lnr-2 eSwNi"
>
  <div
    class="card__Container-sc-s87lnr-1 krKFpC"
  >
    <p
      class="card__Title-sc-s87lnr-0 gdSujY"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-s87lnr-13 gIQTQQ"
    >
      <div
        class="card__Item-sc-s87lnr-12 iMLVzk"
        data-testid="tiger-king-feature"
      >
        <img
          class="card__Image-sc-s87lnr-14 cpMsCx"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 inrCyd"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 igrqGk"
          >
            Tiger King
          </p>
          <p
            class="card__Text-sc-s87lnr-4 gQlijB"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-s87lnr-1 krKFpC"
  >
    <p
      class="card__Title-sc-s87lnr-0 gdSujY"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-s87lnr-13 gIQTQQ"
    >
      <div
        class="card__Item-sc-s87lnr-12 iMLVzk"
        data-testid="juno-feature"
      >
        <img
          class="card__Image-sc-s87lnr-14 cpMsCx"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-s87lnr-11 inrCyd"
        >
          <p
            class="card__SubTitle-sc-s87lnr-3 igrqGk"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-s87lnr-4 gQlijB"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
